packages/mask/src/plugins/Savings/protocols/AAVEProtocol.ts

```diff
@@ -0,0 +1,311 @@
+import type Web3 from 'web3'
+import type { AbiItem } from 'web3-utils'
+import {
+    EthereumTokenType,
+    ChainId,
+    getSavingsConstants,
+    createContract,
+    FungibleTokenDetailed,
+    ZERO_ADDRESS,
+} from '@masknet/web3-shared-evm'
+import type { AaveLendingPool } from '@masknet/web3-contracts/types/AaveLendingPool'
+import type { AaveLendingPoolAddressProvider } from '@masknet/web3-contracts/types/AaveLendingPoolAddressProvider'
+
+import AaveLendingPoolAddressProviderABI from '@masknet/web3-contracts/abis/AaveLendingPoolAddressProvider.json'
+import AaveLendingPoolABI from '@masknet/web3-contracts/abis/AaveLendingPool.json'
+import BigNumber from 'bignumber.js'
+import { ProtocolCategory, SavingsNetwork, SavingsProtocol, ProtocolType } from '../types'
+
+export interface AaveContract {
+    type: EthereumTokenType
+    chainName: string
+    subgraphUrl: string
+    aaveLendingPoolAddressProviderContract: string
+    aaveContract: string
+    stEthContract: string
+}
+
+export const AaveContracts: { [key: number]: AaveContract } = {
+    [ChainId.Mainnet]: {
+        type: EthereumTokenType.ERC20,
+        chainName: 'Ethereum',
+        subgraphUrl: getSavingsConstants(ChainId.Mainnet).AAVE_SUBGRAPHS || '',
+        aaveLendingPoolAddressProviderContract:
+            getSavingsConstants(ChainId.Mainnet).AAVELendingPoolAddressProviderContract || ZERO_ADDRESS,
+        aaveContract: getSavingsConstants(ChainId.Mainnet).AAVE || ZERO_ADDRESS,
+        stEthContract: getSavingsConstants(ChainId.Mainnet).LIDO_STETH || ZERO_ADDRESS,
+    },
+    [ChainId.Gorli]: {
+        type: EthereumTokenType.ERC20,
+        chainName: 'Kovan',
+        subgraphUrl: getSavingsConstants(ChainId.Kovan).AAVE_SUBGRAPHS || '',
+        aaveLendingPoolAddressProviderContract:
+            getSavingsConstants(ChainId.Kovan).AAVELendingPoolAddressProviderContract || ZERO_ADDRESS,
+        aaveContract: getSavingsConstants(ChainId.Kovan).AAVE || ZERO_ADDRESS,
+        stEthContract: getSavingsConstants(ChainId.Kovan).LIDO_STETH || ZERO_ADDRESS,
+    },
+}
+
+export class AAVEProtocol implements SavingsProtocol {
```

Comment:
Does it possible to make this class configurable? Then other token pairs can reuse those methods. After that, all AAVE token pairs can create from configurations.